desktop/cmp/dash/canvas/DashCanvasModel.js

```diff
@@ -313,24 +317,31 @@ export class DashCanvasModel extends HoistModel {
                 title: title ?? viewSpec.title,
                 containerModel: this
             }),
-            prevLayout = previousViewId ? this.getLayout(previousViewId) : null,
+            prevLayout = previousViewId ? this.getViewLayout(previousViewId) : null,
             x = prevLayout?.x ?? layout?.x ?? 0,
             y = prevLayout?.y ?? layout?.y ?? this.rows,
             h = layout?.h ?? viewSpec.height ?? 1,
             w = layout?.w ?? viewSpec.width ?? 1;
 
-        this.layout = [...this.layout, {i: id, x, y, h, w}];
+        this.setLayout([...this.layout, {i: id, x, y, h, w}]);
         this.viewModels = [...this.viewModels, model];
         return model;
     }
 
     @action
     setLayout(layout) {
-        // strip extra properties from react-grid
-        layout = layout.map(({i, x, y, w, h}) => ({i, x, y, w, h}));
-        if (!isEqual(this.layout, layout)) {
-            this.layout = layout;
+        // 1- strip extra properties from react-grid
+        layout = layout.map(({i, x, y, w, h, resizeHandles}) => ({i, x, y, w, h, resizeHandles}));
```

Comment:
Minor thing, but could we use lodash's `pick` here? 
e.g. `layout.map(it => pick(it, {i, x, y, w, h, resizeHandles}))`